Server - initial build of a few components

lfelberg opened the pull request:
Sorry it's long!

lfelberg committed:
Building server-side methods

TODO:
Write HTML/react components for client-side
Connect to server

lfelberg committed:
Building up base client react to serve up

TODO:
build up react side

lfelberg committed:
Getting running on the server

TODO:
Flush out react component
Serve up data

lfelberg committed:
Bare bones of the player are starting to get added in

TODO:
Add in component for waveform
Handle a DB call and send out

lfelberg committed:
Getting to a place where basic react is served up

TODO:
Flush out components on client branch
Get all the errors ironed out

lfelberg committed:
Rendering a facsimile of the site.

TODO:
Add in album art
Complete API request to my server
CSS like crazy!

lfelberg committed:
Adding in SongBar component and Artist names

TODO:
Work on CSS
Render Songbar

gkim795 commented on client/components/SongArt.jsx:
```diff
@@ -0,0 +1,28 @@
+  const handleClick = (e) => {
+    e.preventDefault();
+    const type = (e.target.id === 'player-song-artist') ? 'Artist' : 'Album';
+    handleAlbumClick();
```
If the artist was clicked, would we still want there to be a console log of 'album clicked'?

lfelberg replied:
This is something that I am still trying to figure out how to handle... I am not sure whether I want to have a SongArt method that pops up and alert, or have the method live in Player.jsx. I think it will end up being the latter.

gkim795 commented on client/components/SongBarEntry.jsx:
```diff
@@ -0,0 +1,15 @@
+import PropTypes from 'prop-types';
+
+const SongBarEntry = ({ height, bar }) => (<span
+  className="player-songbar-bar"
```
I would consider bringing the span tag on a new row (or was this an airbnb suggestion?)

lfelberg replied:
Will address this in revisions!

gkim795 commented on client/components/SongBar.jsx:
```diff
@@ -0,0 +1,27 @@
+
+const SongBar = ({ songProfile, handleScan, playtime }) => {
+  const bars = songProfile.profile.map((bar, i) => (<SongBarEntry
+    className="player-songbar-bar"
```
suggestion to move the <SongBarEntry on the next line

gkim795 commented on client/components/SongDisplay.jsx:
```diff
@@ -0,0 +1,39 @@
+    <SongExtras
+      song={song}
+      handleInfoClick={handleInfoClick}
+      id="song-display-extras"
```
stylistic choice of having the id prop listed first or last but try to keep it consistent throughout

lfelberg replied:
Good call I'll make that change to have id come first!

gkim795 commented on client/components/SongDisplay.jsx:
```diff
@@ -0,0 +1,39 @@
+    <SongArt
+      songImage={song.album_imageUrl}
+      handleAlbumClick={handleAlbumClick}
+      id="song-display-album"
```
same as above suggestion

lfelberg replied:
Will address

gkim795 commented on client/components/SongExtras.jsx:
```diff
@@ -0,0 +1,34 @@
+
+  return (
+    <div id="song-display-main-wrapper">
+      <h4 id="player-song-date">{moment(song.added).fromNow()}</h4>
```
is the song.added a timestamp in your database? I tried looking for it in the code and couldn't find it.

lfelberg replied:
It's called song.song_added

gkim795 commented on client/components/SongTracker.jsx:
```diff
@@ -0,0 +1,33 @@
+        songProfile={songProfile}
+        handleScan={handleScan}
+        playtime={playtime}
+        id="player-song-bar"
```
same comment as above with the id placement in props

lfelberg replied:
Will address in revisions!

gkim795 commented on client/components/SongTracker.jsx:
```diff
@@ -0,0 +1,33 @@
+  );
+};
+
+//    <SongComments
```
please remove commented out code before pushing to master

gkim795 reviewed (changes requested):
I had to be really nit-picky to find much to comment on your code. The biggest thing I found was that you included a commented out code. Additionally, there's just some stylistic comments I added.

lfelberg committed:
Small changes to client and going to fix up for pull request

lfelberg committed:
More style changes for code review

Repo kind of a mess, but after these first few implementations,
hopefully I'll have a better hang of git workflow

lfelberg committed:
Adding in changes for pull request

lfelberg committed:
Adding small id refactor for pull req

lfelberg commented:
Changes implemented!
